### ISSUE TITLE
Uncaught UnexpectedValueException: bad in phar://....../psalm.phar/src/Psalm/Internal/Analyzer/Statements/Expression/Call/Method/MethodCallReturnTypeFetcher.php:132

### ISSUE THREAD

**Issue by ArtemGoutsoul** [vimeo/psalm#5183]
Psalm 4.x-dev@0f2a07a9a36d3942da0481f08d4fe89f27d72925 explodes like this (4.4.1 does not, but freezes)

```
Uncaught UnexpectedValueException: bad in phar://..../Psalm/Internal/Analyzer/Statements/Expression/Call/Method/MethodCallReturnTypeFetcher.php:132
Stack trace:
#0 phar://..../Psalm/Internal/Analyzer/Statements/Expression/Call/Method/MethodCallReturnTypeFetcher.php(89): Psalm\Internal\Analyzer\Statements\Expression\Call\Method\MethodCallReturnTypeFetcher::taintMethodCallResult(Object(Psalm\Internal\Analyzer\StatementsAnalyzer), Object(Psalm\Type\Union), Object(_HumbugBoxabc7e063f886\PhpParser\Node\Identifier), Object(_HumbugBoxabc7e063f886\PhpParser\Node\Expr\Variable), Array, Object(Psalm\Internal\MethodIdentifier), Object(Psalm\Internal\MethodIdentifier), 'Nexus_Instance:...', Object(Psalm\Context))
#1 phar://..../Psalm/Internal/Analyzer/Statements/Expression/Call/Method/ExistingAtomicMethodCallAnalyzer.php(100): Psalm\Internal\Analyzer\Statements\Expression\Call\Method\MethodCallReturnTypeFetcher::fetch(Object(Psalm\Internal\Analyzer\StatementsAnalyzer), Object(Psalm\Codebase), Object(_HumbugBoxabc7e063f886\PhpParser\Node\Expr\MethodCall), Object(Psalm\Context), Object(Psalm\Internal\MethodIdentifier), Object(Psalm\Internal\MethodIdentifier), Object(Psalm\Internal\MethodIdentifier), 'Nexus_Instance:...',Object(Psalm\Type\Atomic\TNamedObject), Object(Psalm\Type\Atomic\TNamedObject), Array, Object(Psalm\Internal\Analyzer\Statements\Expression\Call\Method\AtomicMethodCallAnalysisResult), Object(Psalm\Internal\Type\TemplateResult))
#2 phar://..../Psalm/Internal/Analyzer/Statements/Expression/Call/Method/AtomicMethodCallAnalyzer.php(167): Psalm\Internal\Analyzer\Statements\Expression\Call\Method\ExistingAtomicMethodCallAnalyzer::analyze(Object(Psalm\Internal\Analyzer\StatementsAnalyzer), Object(_HumbugBoxabc7e063f886\PhpParser\Node\Expr\MethodCall), Object(_HumbugBoxabc7e063f886\PhpParser\Node\Identifier), Array, Object(Psalm\Codebase), Object(Psalm\Context), Object(Psalm\Type\Atomic\TNamedObject), Object(Psalm\Type\Atomic\TNamedObject), '$this', Object(Psalm\Internal\MethodIdentifier), Object(Psalm\Internal\Analyzer\Statements\Expression\Call\Method\AtomicMethodCallAnalysisResult))
#3 phar://..../Psalm/Internal/Analyzer/Statements/Expression/Call/MethodCallAnalyzer.php(105): Psalm\Internal\Analyzer\Statements\Expression\Call\Method\AtomicMethodCallAnalyzer::analyze(Object(Psalm\Internal\Analyzer\StatementsAnalyzer), Object(_HumbugBoxabc7e063f886\PhpParser\Node\Expr\MethodCall), Object(Psalm\Codebase), Object(Psalm\Context), Object(Psalm\Type\Atomic\TNamedObject), Object(Psalm\Type\Atomic\TNamedObject), false, '$this', Object(Psalm\Internal\Analyzer\Statements\Expression\Call\Method\AtomicMethodCallAnalysisResult))
#4 phar://..../Psalm/Internal/Analyzer/Statements/ExpressionAnalyzer.php(75): Psalm\Internal\Analyzer\Statements\Expression\Call\MethodCallAnalyzer::analyze(Object(Psalm\Internal\Analyzer\StatementsAnalyzer), Object(_HumbugBoxabc7e063f886\PhpParser\Node\Expr\MethodCall), Object(Psalm\Context))
#5 phar://..../Psalm/Internal/Analyzer/Statements/ExpressionAnalyzer.php(32): Psalm\Internal\Analyzer\Statements\ExpressionAnalyzer::handleExpression(Object(Psalm\Internal\Analyzer\StatementsAnalyzer), Object(_HumbugBoxabc7e063f886\PhpParser\Node\Expr\MethodCall), Object(Psalm\Context), false, NULL, true)
#6 phar://..../Psalm/Internal/Analyzer/StatementsAnalyzer.php(360): Psalm\Internal\Analyzer\Statements\ExpressionAnalyzer::analyze(Object(Psalm\Internal\Analyzer\StatementsAnalyzer), Object(_HumbugBoxabc7e063f886\PhpParser\Node\Expr\MethodCall), Object(Psalm\Context), false, NULL, true)
#7 phar://..../Psalm/Internal/Analyzer/StatementsAnalyzer.php(149): Psalm\Internal\Analyzer\StatementsAnalyzer::analyzeStatement(Object(Psalm\Internal\Analyzer\StatementsAnalyzer), Object(_HumbugBoxabc7e063f886\PhpParser\Node\Stmt\Expression), Object(Psalm\Context), NULL)
#8 phar://..../Psalm/Internal/Analyzer/Statements/Block/IfElse/IfAnalyzer.php(44): Psalm\Internal\Analyzer\StatementsAnalyzer->analyze(Array, Object(Psalm\Context))
#9 phar://..../Psalm/Internal/Analyzer/Statements/Block/IfElseAnalyzer.php(188): Psalm\Internal\Analyzer\Statements\Block\IfElse\IfAnalyzer::analyze(Object(Psalm\Internal\Analyzer\StatementsAnalyzer), Object(_HumbugBoxabc7e063f886\PhpParser\Node\Stmt\If_), Object(Psalm\Internal\Scope\IfScope), Object(Psalm\Internal\Scope\IfConditionalScope), Object(Psalm\Context), Object(Psalm\Context), Object(Psalm\Context), Array)
#10 phar://..../Psalm/Internal/Analyzer/StatementsAnalyzer.php(314): Psalm\Internal\Analyzer\Statements\Block\IfElseAnalyzer::analyze(Object(Psalm\Internal\Analyzer\StatementsAnalyzer), Object(_HumbugBoxabc7e063f886\PhpParser\Node\Stmt\If_), Object(Psalm\Context))
#11 phar://..../Psalm/Internal/Analyzer/StatementsAnalyzer.php(149): Psalm\Internal\Analyzer\StatementsAnalyzer::analyzeStatement(Object(Psalm\Internal\Analyzer\StatementsAnalyzer), Object(_HumbugBoxabc7e063f886\PhpParser\Node\Stmt\If_), Object(Psalm\Context), NULL)
#12 phar://..../Psalm/Internal/Analyzer/Statements/Block/LoopAnalyzer.php(117): Psalm\Internal\Analyzer\StatementsAnalyzer->analyze(Array, Object(Psalm\Context))
#13 phar://..../Psalm/Internal/Analyzer/Statements/Block/ForeachAnalyzer.php(171): Psalm\Internal\Analyzer\Statements\Block\LoopAnalyzer::analyze(Object(Psalm\Internal\Analyzer\StatementsAnalyzer), Array, Array, Array, Object(Psalm\Internal\Scope\LoopScope), Object(Psalm\Context), false, false)
#14 phar://..../Psalm/Internal/Analyzer/StatementsAnalyzer.php(326): Psalm\Internal\Analyzer\Statements\Block\ForeachAnalyzer::analyze(Object(Psalm\Internal\Analyzer\StatementsAnalyzer), Object(_HumbugBoxabc7e063f886\PhpParser\Node\Stmt\Foreach_), Object(Psalm\Context))
#15 phar://..../Psalm/Internal/Analyzer/StatementsAnalyzer.php(149): Psalm\Internal\Analyzer\StatementsAnalyzer::analyzeStatement(Object(Psalm\Internal\Analyzer\StatementsAnalyzer), Object(_HumbugBoxabc7e063f886\PhpParser\Node\Stmt\Foreach_), Object(Psalm\Context), Object(Psalm\Context))
#16 phar://..../Psalm/Internal/Analyzer/FunctionLikeAnalyzer.php(254): Psalm\Internal\Analyzer\StatementsAnalyzer->analyze(Array, Object(Psalm\Context), Object(Psalm\Context), true)
#17 phar://..../Psalm/Internal/Analyzer/ClassAnalyzer.php(752): Psalm\Internal\Analyzer\FunctionLikeAnalyzer->analyze(Object(Psalm\Context), Object(Psalm\Internal\Provider\NodeDataProvider), Object(Psalm\Context))
#18 phar://..../Psalm/Internal/Analyzer/ClassAnalyzer.php(211): Psalm\Internal\Analyzer\ClassAnalyzer->analyzeClassMethod(Object(_HumbugBoxabc7e063f886\PhpParser\Node\Stmt\ClassMethod), Object(Psalm\Storage\ClassLikeStorage), Object(Psalm\Internal\Analyzer\ClassAnalyzer), Object(Psalm\Context), Object(Psalm\Context))
#19 phar://..../Psalm/Internal/Analyzer/FileAnalyzer.php(170): Psalm\Internal\Analyzer\ClassAnalyzer->analyze(Object(Psalm\Context), Object(Psalm\Context))
#20 phar://..../Psalm/Internal/Codebase/Analyzer.php(267): Psalm\Internal\Analyzer\FileAnalyzer->analyze(NULL)
#21 phar://..../Psalm/Internal/Codebase/Analyzer.php(418): Psalm\Internal\Codebase\Analyzer->Psalm\Internal\Codebase\{closure}(67, 'C:\\dev\\nexus.gi...')
#22 phar://..../Psalm/Internal/Codebase/Analyzer.php(222): Psalm\Internal\Codebase\Analyzer->doAnalysis(Object(Psalm\Internal\Analyzer\ProjectAnalyzer), 1)
#23 phar://..../Psalm/Internal/Analyzer/ProjectAnalyzer.php(447): Psalm\Internal\Codebase\Analyzer->analyzeFiles(Object(Psalm\Internal\Analyzer\ProjectAnalyzer), 1, false, true)
#24 phar://..../psalm.php(469): Psalm\Internal\Analyzer\ProjectAnalyzer->check('C:\\dev\\nexus.gi...', false)
#25 phar://C:/dev/nexus.git/psalm.phar/psalm(5): require_once('phar://C:/dev/n...')
#26 C:\dev\nexus.git\psalm.phar(14): require('phar://C:/dev/n...')
#27 {main}
(Psalm 4.x-dev@0f2a07a9a36d3942da0481f08d4fe89f27d72925 crashed due to an uncaught Throwable)
```

**Comment by orklah** [comment]
I think this was fixed in https://github.com/vimeo/psalm/commit/7dbdc8c59e619adb70a766faa458ee0b2c0a239e 4 commits later than your version

**Comment by ArtemGoutsoul** [comment]
Thanks @orklah! I'll try again with a phar built from that commit

**Comment by ArtemGoutsoul** [comment]
@orklah Worked! thank you!

**Comment by orklah** [comment]
Great! Can you close the ticket? :)

**Comment by ArtemGoutsoul** [comment]
As @orklah pointed out, this issue was resolved in 7dbdc8c
